app.py

```diff
@@ -0,0 +1,139 @@
+from flask import Flask, request, jsonify, make_response
+from flask_sqlalchemy import SQLAlchemy
+from os import environ
+from abc import ABC, abstractmethod
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = environ.get('DATABASE_URL')
+db = SQLAlchemy(app)
+
+
+class User(db.Model):
+    __tablename__ = 'users'
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(30), unique=False, nullable=False)
+    email = db.Column(db.String(50), unique=True, nullable=False)
+
+    def json(self):
+        return {
+            'id': self.id,
+            'username': self.username,
+            'email': self.email
+        }
+
+
+class AbstractDatabaseAdapter(ABC):
+
+    @abstractmethod
+    def create(self, entity):
+        pass
+
+    @abstractmethod
+    def get_all(self, entity_type):
+        pass
+
+    @abstractmethod
+    def get_by_id(self, entity_type, entity_id):
+        pass
+
+    @abstractmethod
+    def update(self, entity):
+        pass
+
+    @abstractmethod
+    def delete(self, entity):
+        pass
+
+
+class PostgreSQLAdapter(AbstractDatabaseAdapter):
+
+    def create(self, entity):
+        db.session.add(entity)
+        db.session.commit()
+
+    def get_all(self, entity_type):
+        return entity_type.query.all()
+
+    def get_by_id(self, entity_type, entity_id):
+        return entity_type.query.get(entity_id)
+
+    def update(self, entity):
+        db.session.commit()
+
+    def delete(self, entity):
+        db.session.delete(entity)
+        db.session.commit()
+
+
+db_adapter = PostgreSQLAdapter()
+
+with app.app_context():
+    db.create_all()
+
+
+@app.route('/test', methods=['GET'])
+def test():
+    return make_response(jsonify({'message': 'test route message'}), 200)
+
+
+@app.route('/user', methods=['POST'])
+def create_user():
+    try:
+        data = request.get_json()
+        new_user = User(username=data['username'], email=data['email'])
+        db_adapter.create(new_user)
+        return make_response(jsonify({'message': 'user created'}), 201)
+    except Exception as e:
+        return make_response(jsonify({'message': 'error creating user'}), 500)
+
+
+@app.route('/users', methods=['GET'])
+def get_users():
+    try:
+        users = db_adapter.get_all(User)
+        return make_response(jsonify([user.json() for user in users]), 200)
```

Comment:
you can try to optimize the response here, what if there are 10000 users in the db? Probably pagination

Reply:
I added pagination